<launch>

	<param name="/auvsi16/server_ip" value="10.0.2.1" />
	<param name="/auvsi16/server_port" value="8080" />
	<param name="/auvsi16/team_code" value="AMVUI" />
	<param name="/auvsi16/course_type" value="courseA" />
	<param name="/auvsi16/video/right/resize/compressed/jpeg_quality" value="10" />
	<param name="/auvsi16/video/front/resize/compressed/jpeg_quality" value="10" />

<!--	<node name="video_feed_upstream" pkg="auvsi16" type="video_feed_upstream" output="screen">
		<param name="resize_width_factor" value="0.3" />
		<param name="resize_height_factor" value="0.3" />
	</node> -->

	<node name="video_feed_front" pkg="auvsi16" type="video_feed_front" output="screen">
		<param name="camera_port" value="1" />
		<param name="camera_width" value="1280" />
		<param name="camera_height" value="720" />

		<param name="v4l2_configuration" value="0" />
	</node>

	<node name="video_feed_right" pkg="auvsi16" type="video_feed_right" output="screen">
		<param name="camera_port" value="0" />
		<param name="camera_width" value="1280" />
		<param name="camera_height" value="720" />

		<param name="v4l2_configuration" value="0" />
	</node>

	<!-- PID Here is deprecated change in the pid_controller  -->
	<!-- Change PID in the code by publishing to pid_const-->

	<arg name="pid_rate" default="100" />

	<arg name="kp" default="0" />
	<arg name="ki" default="0" />
	<arg name="kd" default="0" />
	<arg name="upper_limit" default="2000" />
	<arg name="lower_limit" default="-2000" />

	<!-- anti-windup is to clap integral value -->
	<arg name="anti_windup" default="0.1" />

	<arg name="topic_pid_in" default="auvsi16/pid/in" />
	<arg name="topic_pid_out" default="auvsi16/pid/out" />

	<node name="pid_controller" pkg="pid" output="screen" type="controller_adaptive" args="$(arg kp) $(arg ki) $(arg kd) $(arg pid_rate) -tfp $(arg topic_pid_in) -tfc $(arg topic_pid_out) -ul $(arg upper_limit) -ll $(arg lower_limit) -aw $(arg anti_windup)" />

	<node name="override_monitor" pkg="auvsi16" output="screen" type="override_monitor"/>
	<node name="motor_controller" pkg="auvsi16" output="screen" type="motor_controller"/>

	<node name="navigation_mission_control" pkg="auvsi16" output="screen" type="navigation_mission_control" >
			<param name="arena_distance" value="46" />
			<param name="first_lat" value="-6.367437" />
			<param name="first_long" value="106.829882" />
			<param name="second_lat" value="-6.367437" />
			<param name="second_long" value="106.829882" />
	</node>
	<node name="docking_mission_control" pkg="auvsi16" output="screen" type="docking_mission_control">
		<param name="heading_zero" value="49" />
		<param name="center_dock_lat" value="-6.367437" />
		<param name="center_dock_long" value="106.829882" />
		<param name="distance_from_shape" value="7" />
		<param name="shape_arena_range" value="20" />
		<param name="distance_to_dock" value="2" />
		<param name="distance_to_away" value="10" />
		<param name="detection_accuracy" value="15" />
	</node>

	<node name="interoperability_mission_control_simple" pkg="auvsi16" output="screen" type="interoperability_mission_control_simple">
		<param name="heading_to_deploy_auv" value="300" />
		<param name="entrance_lat" value="36.8017223" />
		<param name="entrance_long" value="-76.1926523" />
	</node>
	<node name="heartbeat" pkg="auvsi16" type="heartbeat" />
	<node name="run_course" pkg="auvsi16" output="screen" type="run_course" />

	<!--
	<node name="obstacle_mission_control_simple" pkg="auvsi16" output="screen" type="obstacle_mission_control_simple">
		<param name="obstacle_heading" value="10" />
		<param name="arena_distance" value="40" />
	</node>
	-->

</launch>
